smarts/core/bubble_manager.py

```diff
@@ -266,93 +312,39 @@ def step(self):
-
-        for vehicle_id, actor in state_change.entered_airlock_1:
-            self._airlock_social_vehicle_with_social_agent(vehicle_id, actor)
-
-        for vehicle_id, actor in state_change.entered_bubble:
-            self._hijack_social_vehicle_with_social_agent(vehicle_id, actor)
-
-        # XXX: Some vehicles only go through the airlocks and never make it through
-        #      the bubble; that's why we relinquish on airlock exit. This is something
-        #      we'll likely want to revisit in the future.
-        for vehicle_id in state_change.exited_airlock_2:
-            self._relinquish_vehicle_to_traffic_sim(vehicle_id)
-
-    def step_bubble_state(
-        self,
-        social_vehicles: Sequence[Vehicle],
-        social_agent_vehicles: Sequence[Vehicle],
-    ) -> BubbleStateChange:
-        self._move_travelling_bubbles()
-
-        # Detect social vehicles entering bubbles (and airlocks)
-        for sv in social_vehicles:
-            cursor = self._find_cursor(sv.id)
-            if cursor is not None:
-                # Already tracking this vehicle (in airlock)
-                continue
-
-            # We didn't have a cursor for this vehicle yet
-            cursor = self._obtain_cursor_if_entered_airlock(sv)
-            if cursor is not None:
-                self._cursors.append(cursor)
 
-        kept_cursors = []
-        entered_airlock_1, entered_bubble = [], []
-        exited_bubble, exited_airlock_2 = [], []
+        transitioned = [c for c in cursors if c.transition is not None]
+        for cursor in transitioned:
+            is_social = cursor.tracking_id in sim.vehicle_index.social_vehicle_ids
+            is_agent = cursor.tracking_id in sim.vehicle_index.agent_vehicle_ids
 
-        for cursor in self._cursors:
-            if self._cursor_points_to_destroyed_vehicle(
-                cursor, social_vehicles, social_agent_vehicles
-            ):
-                self._log.debug(
-                    f"cursor={cursor.tracking_id} is pointing to a destroyed vehicle; "
-                    "dropping it"
+            if is_social and cursor.transition == BubbleTransition.AirlockEntered:
+                self._airlock_social_vehicle_with_social_agent(
+                    sim, cursor.tracking_id, cursor.bubble.actor
                 )
-                continue
-
-            if not cursor.update():
-                kept_cursors.append(cursor)
-                continue
-
-            if cursor.state == BubbleState.AirlockExited:
-                self._log.debug(
-                    f"vehicle={cursor.tracking_id} exited bubble and airlock"
+            elif is_social and cursor.transition == BubbleTransition.Entered:
+                self._hijack_social_vehicle_with_social_agent(
+                    sim, cursor.tracking_id, cursor.bubble.actor
                 )
-                exited_airlock_2.append(cursor.tracking_id)
+            # TODO: Diff approach presently has no way of knowing if this was a
```

Comment:
This is real problem w/ this approach that actually require us to store (some) state...

Reply:
From discussion,
* Could solve this by storing the fact that we were hijacked in the `VehicleIndex`